### PR TITLE
[SourceMap] Add unit tests

### PR DESCRIPTION
**Pull request by jasonLaster**
This adds a unit test to the source map package. 

It also moves the worker code to a `source-map` module so that it is easier to separate the worker plumbing from the implementation logic. See [parser/worker](https://github.com/devtools-html/debugger.html/blob/master/src/utils/parser/worker.js).

### CODE REVIEW

File: packages/devtools-modules/client/shared/worker.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// @flow
+
+export type Message = {
```
Comment (jasonLaster) [inline packages/devtools-modules/client/shared/worker.js]: I added the worker module so that it can be shared.

File: packages/devtools-source-map/.babelrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "plugins": [
+    "transform-async-to-generator",
+    "transform-flow-strip-types"
+  ]
+}
```
Comment (jasonLaster) [inline packages/devtools-source-map/.babelrc]: it's helpful to ask babel to transform modules here too

File: packages/devtools-source-map/src/worker.js [hunk 3]
```diff
@@ -1,234 +1,19 @@
...
-  const map = SourceMapConsumer(generator.toJSON());
-  sourceMapRequests.set(generatedId, Promise.resolve(map));
-}
-
-const publicInterface = {
   getOriginalURLs,
   getGeneratedLocation,
   getOriginalLocation,
   getOriginalSourceText,
-  enableSourceMaps,
```
Comment (jryans) [inline packages/devtools-source-map/src/worker.js]: Hmm, I am a bit perplexed by `enableSourceMaps`... It doesn't look like debugger is calling it even today? So... how are source maps working...? 😕 Are you removing it because it's unused?

File: packages/devtools-source-map/src/tests/source-map.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const { getOriginalURLs } = require("../source-map");
+
+function getMap(_path) {
+  const mapPath = path.join(__dirname, _path);
+  return fs.readFileSync(mapPath, "utf8");
+}
+
+describe("source maps", () => {
+  test("getOriginalURLs", async () => {
+    const source = {
```
Comment (jryans) [inline packages/devtools-source-map/src/tests/source-map.js]: Would it be possible to check in the original sources of the bundle and a script for bundling them? There could be cases where we need to fix issues with source map _generation_, so being able to regenerate test bundles in a sane way seems useful.

File: packages/devtools-source-map/src/worker.js (hunk 3 above)
Comment (jasonLaster) [inline packages/devtools-source-map/src/worker.js]: It's used in restartWorker - but it's too complicated and we should be feature flagging higher up

File: packages/devtools-source-map/src/tests/source-map.js (hunk 4 above)
Comment (jasonLaster) [inline packages/devtools-source-map/src/tests/source-map.js]: Hmm, that makes sense. 

I think for this test it's fine that we have a simple map file. For other tests it'd be nice to have webpack / browserify etc

File: packages/devtools-modules/client/shared/worker.js (hunk 1 above)
Comment (jasonLaster) [inline packages/devtools-modules/client/shared/worker.js]: Hmm - I agree, this is more than just unit tests. Is it okay to do in a separate commit?

File: packages/devtools-source-map/src/worker.js (hunk 3 above)
Comment (jryans) [inline packages/devtools-source-map/src/worker.js]: Ah, thanks for the pointer. Seems fine to move up to a higher level. Shouldn't we update source-map's `index.js` to stop sending the `enableSourceMap` message then?

File: packages/devtools-source-map/src/worker.js [hunk 5]
```diff
@@ -1,223 +1,17 @@
...
-
-function applySourceMap(generatedId, url, code, mappings) {
-  const generator = new SourceMapGenerator({ file: url });
-  mappings.forEach(mapping => generator.addMapping(mapping));
-  generator.setSourceContent(url, code);
-
-  const map = SourceMapConsumer(generator.toJSON());
-  sourceMapRequests.set(generatedId, Promise.resolve(map));
-}
+  getOriginalURLs,
```
Comment (jryans) [inline packages/devtools-source-map/src/worker.js]: Please add a comment somewhere in this file to explain that the indirection to another module is for testing the worker's logic without actually launching a worker.

Comment (jasonLaster) [inline packages/devtools-source-map/src/worker.js]: sure

File: packages/devtools-source-map/.babelrc [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "plugins": [
+    "transform-async-to-generator",
```
Comment (jasonLaster) [inline packages/devtools-source-map/.babelrc]: Depends on the node version :)

Also we might still want generators for perf

**Comment by jryans** [comment]
I guess a good next step is to run the tests on CI as well? (I don't think we're running `jest` there for this package yet.)

**Comment by jryans** [comment]
My last two review comments appear unresolved in the latest commit, so let me know when it's ready for another look. 😁

**Comment by jryans** [comment]
@jasonLaster, seems like following review comments are still left:

[stop sending `enableSourceMap` message](https://github.com/devtools-html/devtools-core/pull/255#discussion_r106714836)
[add a comment about worker indirection](https://github.com/devtools-html/devtools-core/pull/255#discussion_r106715280)

**Comment by jasonLaster** [comment]
Thanks @jryans I left those changes on my local branch :/
